wkcuber/__main__.py

```diff
@@ -36,52 +57,49 @@ def create_parser() -> ArgumentParser:
 def main(args: Namespace) -> None:
     setup_logging(args)
 
-    bounding_box = cubing(
-        args.source_path,
-        args.target_path,
-        args.layer_name,
-        args.dtype,
-        args.batch_size,
-        args,
-    )
+    auto_detect_and_run_conversion(args)
 
-    write_webknossos_metadata(
-        args.target_path,
-        args.name,
-        args.scale,
-        compute_max_id=False,
-        exact_bounding_box=bounding_box,
-    )
+    layer_path_to_mags: Dict[Path, List[Mag]] = detect_present_mags(args.target_path)
 
     if not args.no_compress:
-        compress_mag_inplace(args.target_path, args.layer_name, Mag(1), args)
+        for (layer_path, mags) in layer_path_to_mags.items():
+            layer_name = layer_path.stem
+            for mag in mags:
+                compress_mag_inplace(args.target_path, layer_name, mag, args)
 
     if not args.isotropic:
-        downsample_mags_anisotropic(
-            args.target_path,
-            args.layer_name,
-            Mag(1),
-            Mag(args.max_mag),
-            args.scale,
-            "default",
-            not args.no_compress,
-            args=args,
-        )
+        for (layer_path, mags) in layer_path_to_mags.items():
+            layer_name = layer_path.stem
+            mags.sort(reverse=True)
+            downsample_mags_anisotropic(
+                args.target_path,
+                layer_name,
+                mags[0],
```

Comment:
shouldn't it be this?
```suggestion
                mags[-1],
```

Otherwise, mag 1 would overwrite the existing mag 2 if both exist? Also: Is the mags list natsorted? This would be necessary for this to work.

Reply:
Ah, I missed line 73. Alrighty then. I'd prefer to not use `reverse=True` and instead access `[-1]`. But you can leave it as is.

However: Wouldn't the sorting need to be natsorted? For example, sorting `[1, 8, 16]` wouldn't work otherwise, I think.

Reply:
As far as I tested it, it works. My guess would be that the `Mag` objects that we're using have an internal correct sorting method (or less-equal method). I went with your suggestion and accessed `[-1]`

Reply:
> My guess would be that the Mag objects that we're using have an internal correct sorting method (or less-equal method).

Ah, you are right! I thought of them as strings, but that was wrong :) The class indeed implements `__le__` and co :thinking: